src/extension-support/extension-manager.js

```diff
@@ -171,16 +155,25 @@ class ExtensionManager {
      * @returns {Promise} resolved once all the extensions have been reinitialized
      */
     refreshBlocks () {
-        const allPromises = Array.from(this._loadedExtensions.values()).map(serviceName =>
-            dispatch.call(serviceName, 'getInfo')
+        const loadedExtensions = this.runtime.getLoadedExtensions();
+        const allPromises = [];
+        loadedExtensions.forEach((extensionInfo, oldId) => {
+            const serviceName = extensionInfo.serviceName;
+            allPromises.push(dispatch.call(serviceName, 'getInfo')
                 .then(info => {
                     info = this._prepareExtensionInfo(serviceName, info);
+                    loadedExtensions.set(info.id, info);
+                    if (oldId !== info.id) {
+                        log.warn(`Extension changed ID from ${oldId} to ${info.id}!`);
```

Comment:
Hmmm is this still relevant now that the entire extension manager is no longer being moved to the runtime in this PR? There shouldn't really be a way to change the extension ID right?

Reply:
Note from conversation with @kchadha: it's probably best to keep `oldId` instead of switching to `info.id` here.